Source/WebGPU/WGSL/Parser.cpp

```diff
@@ -329,6 +332,45 @@ Expected<UniqueRef<AST::TypeDecl>, Error> Parser<Lexer>::parseTypeDeclAfterIdent
     RETURN_NODE_REF(NamedType, WTFMove(name));
 }
 
+// https://www.w3.org/TR/WGSL/#syntax-array_type_decl
+// ArrayTypeDecl:
+//      'array' ('<' TypeDecl (',' ElementCountExpression)? '>')?
+// ElementCountExpression:
+//      AdditiveExpression | BitwiseExpression
+template<typename Lexer>
+Expected<UniqueRef<AST::TypeDecl>, Error> Parser<Lexer>::parseArrayType()
```

Comment:
I guess I already said this in a previous comment, but I'm super surprised we have custom parsing logic for certain types and not others. Seems more elegant to just have a generic parsing function, with full flexibility, and then let type checking figure out what's an error.

I understand that the spec has custom parsing logic for certain types, but does that mean we need to follow the same practice? We only need to behave "as-if" we followed the spec.